applications/HDF5Application/custom_io/hdf5_model_part_io.cpp

```diff
@@ -1,59 +1,77 @@
-#include "utilities/builtin_timer.h"
-#include "utilities/openmp_utils.h"
-#include "input_output/logger.h"
+#include "custom_io/hdf5_properties_io.h"
 #include "custom_utilities/container_io_utils.h"
+#include "custom_utilities/factor_elements_and_conditions_utility.h"
+#include "custom_utilities/hdf5_data_set_partition_utility.h"
+#include "hdf5_application_variables.h"
+
+// Include base h
+#include "custom_io/hdf5_model_part_io.h"
 
 namespace Kratos
 {
 namespace HDF5
 {
-namespace
+ModelPartIO::ModelPartIO(
+    Parameters Settings,
+    File::Pointer pFile)
+    : mpFile(pFile),
+      mWriteEntityProperyIds(true)
 {
-template <typename TContainer>
-void WriteContainerIds(File& rFile, std::string const& rPath, TContainer const& rContainer, WriteInfo& rInfo)
-{
-    KRATOS_TRY;
-    Vector<int> ids;
-    ids.resize(rContainer.size());
-    #pragma omp parallel for
-    for (int i = 0; i < static_cast<int>(rContainer.size()); ++i)
-    {
-        const auto it = rContainer.begin() + i;
-        ids[i] = it->Id();
-    }
-    rFile.WriteDataSet(rPath, ids, rInfo);
-    KRATOS_CATCH("");
-}
+    Parameters default_params(R"(
+        {
+            "prefix"           : "",
+            "custom_attributes": {}
+        })");
 
-int GlobalNumberOfNodes(ModelPart const& rModelPart)
-{
-    int number_of_nodes = rModelPart.NumberOfNodes();
-    return rModelPart.GetCommunicator().GetDataCommunicator().SumAll(number_of_nodes);
-}
+    Settings.AddMissingParameters(default_params);
+    mPrefix = Settings["prefix"].GetString();
 
-int GlobalNumberOfElements(ModelPart const& rModelPart)
-{
-    int number_of_elems = rModelPart.NumberOfElements();
-    return rModelPart.GetCommunicator().GetDataCommunicator().SumAll(number_of_elems);
-}
+    mCustomAttributes = Settings["custom_attributes"];
 
-int GlobalNumberOfConditions(ModelPart const& rModelPart)
-{
-    int number_of_conds = rModelPart.NumberOfConditions();
-    return rModelPart.GetCommunicator().GetDataCommunicator().SumAll(number_of_conds);
-}
+    KRATOS_ERROR_IF(mCustomAttributes.Has("__model_part_name"))
+        << "The \"__model_part_name\" is a reserved attribute name. Please remove it from custom attributes."
+        << " Custom attributes:\n" << mCustomAttributes << "\n";
+
+    KRATOS_ERROR_IF(mPrefix.empty()) << "ModelPartIO requires non-empty prefix.\n";
+
+    KRATOS_ERROR_IF(mPrefix.back() == '/')
+        << "The ModelPartIO prefix is always assumed to be representing a group,"
+        << " hence ending \"\\\" is not required [ provided prefix = "
+        << mPrefix << " ].\n";
 }
 
-ModelPartIO::ModelPartIO(File::Pointer pFile, std::string const& rPrefix)
-: mpFile(pFile), mPrefix(rPrefix)
+ModelPartIO::ModelPartIO(
+    const std::string& rPrefix,
+    const bool WriteEntityPropertyIds,
+    File::Pointer pFile)
+    : mpFile(pFile),
+      mPrefix(rPrefix),
+      mWriteEntityProperyIds(WriteEntityPropertyIds)
 {
 }
```

Comment:
The two constructors are not equivalent because this one doesn't do any input validation. I recommend delegating to the other constructor.

E.g.: smth like

```suggestion
ModelPartIO::ModelPartIO(
    const std::string& rPrefix,
    const bool WriteEntityPropertyIds,
    File::Pointer pFile)
    : ModelPartIO(
        [&rPrefix](){Parameters output; output.AddString("prefix", rPrefix); return output}(),
        pFile)
{
    mWriteEntityProperyIds = WriteEntityPropertyIds;
}
```

Reply:
done :)